<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>650</width>
    <height>500</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout">
    <item>
     <widget class="QTabWidget" name="tabWidget">
      <property name="currentIndex">
       <number>0</number>
      </property>
      <widget class="QWidget" name="textConvTab">
       <attribute name="title">
        <string>Text / Base64 / URL</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout_2">
        <item>
         <widget class="QLabel" name="label">
          <property name="text">
           <string>&lt;b&gt;Input&lt;/b&gt;</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPlainTextEdit" name="textInput">
          <property name="font">
           <font>
            <family>Consolas</family>
            <pointsize>10</pointsize>
           </font>
          </property>
         </widget>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout">
          <item>
           <widget class="QPushButton" name="btnTextToHex">
            <property name="text">
             <string>Text → Hex</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="btnHexToText">
            <property name="text">
             <string>Hex → Text</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="btnTextToBase64">
            <property name="text">
             <string>Text → Base64</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="btnBase64ToText">
            <property name="text">
             <string>Base64 → Text</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="btnUrlEncode">
            <property name="text">
             <string>URL Encode</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="btnUrlDecode">
            <property name="text">
             <string>URL Decode</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <widget class="QLabel" name="label_2">
          <property name="text">
           <string>&lt;b&gt;Output&lt;/b&gt;</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPlainTextEdit" name="textOutput">
          <property name="font">
           <font>
            <family>Consolas</family>
            <pointsize>10</pointsize>
           </font>
          </property>
          <property name="readOnly">
           <bool>true</bool>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="hashTab">
       <attribute name="title">
        <string>Hashing (MD5, SHA)</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout_3">
        <item>
         <widget class="QLabel" name="label_3">
          <property name="text">
           <string>&lt;b&gt;Input Text&lt;/b&gt; (Hashes will update automatically as you type)</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPlainTextEdit" name="hashInputText"/>
        </item>
        <item>
         <layout class="QFormLayout" name="formLayout">
          <item row="0" column="0">
           <widget class="QLabel" name="label_4">
            <property name="text">
             <string>&lt;b&gt;MD5:&lt;/b&gt;</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QLineEdit" name="md5Result">
            <property name="font">
             <font>
              <family>Consolas</family>
             </font>
            </property>
            <property name="readOnly">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="label_5">
            <property name="text">
             <string>&lt;b&gt;SHA-1:&lt;/b&gt;</string>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QLineEdit" name="sha1Result">
            <property name="font">
             <font>
              <family>Consolas</family>
             </font>
            </property>
            <property name="readOnly">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item row="2" column="0">
           <widget class="QLabel" name="label_6">
            <property name="text">
             <string>&lt;b&gt;SHA-256:&lt;/b&gt;</string>
            </property>
           </widget>
          </item>
          <item row="2" column="1">
           <widget class="QLineEdit" name="sha256Result">
            <property name="font">
             <font>
              <family>Consolas</family>
             </font>
            </property>
            <property name="readOnly">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item row="3" column="0">
           <widget class="QLabel" name="label_7">
            <property name="text">
             <string>&lt;b&gt;SHA-512:&lt;/b&gt;</string>
            </property>
           </widget>
          </item>
          <item row="3" column="1">
           <widget class="QLineEdit" name="sha512Result">
            <property name="font">
             <font>
              <family>Consolas</family>
             </font>
            </property>
            <property name="readOnly">
             <bool>true</bool>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <spacer name="verticalSpacer">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>40</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="numberBaseTab">
       <attribute name="title">
        <string>Number Base Converter</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout_4">
        <item>
         <widget class="QLabel" name="label_12">
          <property name="text">
           <string>Enter a value in any field and the others will update automatically.</string>
          </property>
         </widget>
        </item>
        <item>
         <layout class="QFormLayout" name="formLayout_2">
          <item row="0" column="0">
           <widget class="QLabel" name="label_8">
            <property name="text">
             <string>&lt;b&gt;Decimal&lt;/b&gt;</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QLineEdit" name="decLineEdit">
            <property name="font">
             <font>
              <family>Consolas</family>
              <pointsize>10</pointsize>
             </font>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="label_9">
            <property name="text">
             <string>&lt;b&gt;Hexadecimal&lt;/b&gt;</string>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QLineEdit" name="hexLineEdit">
            <property name="font">
             <font>
              <family>Consolas</family>
              <pointsize>10</pointsize>
             </font>
            </property>
           </widget>
          </item>
          <item row="2" column="0">
           <widget class="QLabel" name="label_10">
            <property name="text">
             <string>&lt;b&gt;Binary&lt;/b&gt;</string>
            </property>
           </widget>
          </item>
          <item row="2" column="1">
           <widget class="QLineEdit" name="binLineEdit">
            <property name="font">
             <font>
              <family>Consolas</family>
              <pointsize>10</pointsize>
             </font>
            </property>
           </widget>
          </item>
          <item row="3" column="0">
           <widget class="QLabel" name="label_11">
            <property name="text">
             <string>&lt;b&gt;Octal&lt;/b&gt;</string>
            </property>
           </widget>
          </item>
          <item row="3" column="1">
           <widget class="QLineEdit" name="octLineEdit">
            <property name="font">
             <font>
              <family>Consolas</family>
              <pointsize>10</pointsize>
             </font>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <spacer name="verticalSpacer_2">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>196</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="timestampTab">
       <attribute name="title">
        <string>Unix Timestamp</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout_5">
        <item>
         <layout class="QFormLayout" name="formLayout_3">
          <item row="0" column="0">
           <widget class="QLabel" name="label_13">
            <property name="text">
             <string>&lt;b&gt;Unix Timestamp (seconds)&lt;/b&gt;</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QLineEdit" name="timestampLineEdit">
            <property name="font">
             <font>
              <family>Consolas</family>
              <pointsize>10</pointsize>
             </font>
            </property>
           </widget>
          </item>
          <item row="2" column="0">
           <widget class="QLabel" name="label_14">
            <property name="text">
             <string>&lt;b&gt;Human-Readable Date (UTC)&lt;/b&gt;</string>
            </property>
           </widget>
          </item>
          <item row="2" column="1">
           <widget class="QDateTimeEdit" name="dateTimeEdit">
            <property name="font">
             <font>
              <pointsize>10</pointsize>
             </font>
            </property>
            <property name="displayFormat">
             <string>yyyy-MM-dd hh:mm:ss</string>
            </property>
            <property name="calendarPopup">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <layout class="QHBoxLayout" name="horizontalLayout_2">
            <item>
             <widget class="QPushButton" name="btnTimestampToDate">
              <property name="text">
               <string>▼   Convert to Date   ▼</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="btnDateToTimestamp">
              <property name="text">
               <string>▲ Convert to Timestamp ▲</string>
              </property>
             </widget>
            </item>
            <item>
             <spacer name="horizontalSpacer">
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>40</width>
                <height>20</height>
               </size>
              </property>
             </spacer>
            </item>
            <item>
             <widget class="QPushButton" name="btnNow">
              <property name="text">
               <string>Now</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </item>
        <item>
         <spacer name="verticalSpacer_3">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>298</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>650</width>
     <height>21</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
